homeassistant/components/v2c/sensor.py

```diff
@@ -0,0 +1,278 @@
+    ),
+    SensorEntityDescription(
+        key=HOUSE_POWER_KEY,
+        name="House Power Trydan",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key=FV_POWER_KEY,
+        name="PV Power Trydan",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key=PAUSE_STATE_READ_KEY,
+        name="Pause State Read",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=LOCK_READ_KEY,
+        name="Lock Read",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=PROMGRAM_READ_KEY,
+        name="Promgram Read",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=INTENSITY_READ_KEY,
+        name="Intensity Read",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=DYNAMIC_READ_KEY,
+        name="Dynamic Read",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=PAYMENT_READ_KEY,
+        name="Payment Read",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=OCPP_READ_KEY,
+        name="Ocpp Read",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=PAUSE_STATE_KEY,
+        name="Pause state",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=LOCK_KEY,
+        name="Lock state Trydan",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=PROMGRAM_KEY,
+        name="Promgram state Trydan",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=INTENSITY_KEY,
+        name="Trydan Intensity",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key=DYNAMIC_KEY,
+        name="Trydan Dynamic",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=PAYMENT_KEY,
+        name="Trydan payment",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=OCPP_KEY,
+        name="OCPP state Trydan",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+SENSOR_KEYS: list[str] = [desc.key for desc in TRYDAN_SENSOR_TYPES]
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_MONITORED_VARIABLES, default=["status"]): vol.All(
+            cv.ensure_list, [vol.In(SENSOR_KEYS)]
+        ),
+    }
+)
```

Comment:
We no longer allow integrations to add or change a platform YAML configuration.

More information on this can be found in Architecture Decision Record:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

Please note that this integration connects to a device or service and another Architecture Decision Record applies that disallows the use of YAML configuration in favor of a configuration flow via the UI:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

As these changes often involve a bit of work and some significant shift in the current code, we will close this PR for now.

Thanks already! :+1:

Reply:
Thanks for the info, i will change it.